packages/smart-wallet/src/smartWallet.js

```diff
@@ -745,6 +757,38 @@ export const prepareSmartWallet = (baggage, shared) => {
           }
           return baggage.get(state.address);
         },
+        /** @param {string} suggestion */
+        findUnusedName(suggestion) {
+          const { myStore } = this.state;
+          let nonce = 0;
+          let name = suggestion;
+          while (myStore.has(name)) {
+            nonce += myStore.getSize(); // avoid linear work
+            name = `${suggestion}.${nonce}`;
+          }
+          return name;
+        },
+        /**
+         * @param {ResultPlan} plan
+         * @param {unknown} result
+         */
+        saveOfferResult(plan, result) {
```

Comment:
This isn't just offer results now. And it doesn't always save.

Consider `maybeSaveResult`

Reply:
Sorry, partial attention during leave: if there's a ResultPlan doesn't that imply saving?

Rather than calling a maybe, the saveEntry should only be called when there's a plan. Right?